clients/client-java/src/main/java/org/apache/gravitino/client/GenericRole.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.client;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.apache.gravitino.Audit;
+import org.apache.gravitino.authorization.Role;
+import org.apache.gravitino.authorization.SecurableObject;
+import org.apache.gravitino.dto.authorization.RoleDTO;
+
+/** Represents a generic role. This class is used for lazily requesting securable objects. */
+class GenericRole implements Role {
+
+  private final RoleDTO roleDTO;
+  private final Supplier<List<SecurableObject>> securableObjectsSupplier;
+
+  GenericRole(RoleDTO roleDTO, GravitinoMetalake gravitinoMetalake) {
+    this.roleDTO = roleDTO;
+    this.securableObjectsSupplier =
+        new Supplier<List<SecurableObject>>() {
+          private boolean waitToRequest = true;
+          private List<SecurableObject> securableObjects;
+
+          @Override
+          public List<SecurableObject> get() {
+            if (waitToRequest) {
+              securableObjects = gravitinoMetalake.getRole(roleDTO.name()).securableObjects();
```

Comment:
I think you can have a new API to get securable objects, rather than using get role API, you can follow what Tag did.

Reply:
I ever thought. But from my ' view, our role API can already get the objects. It's a little weird to add a new API to fetch partial data.

Reply:
It's weird to get a role in a role from client side.